azurelinuxagent/common/protocol/extensions_goal_state.py

```diff
@@ -108,8 +109,10 @@ def compare_attribute(attribute):
         compare_attribute("activity_id")
         compare_attribute("correlation_id")
         compare_attribute("created_on_timestamp")
-        compare_attribute("status_upload_blob")
-        compare_attribute("status_upload_blob_type")
+        # The status blob was added after version 112
```

Comment:
I'm guessing this check is temporary until this feature is stabilized? If so, could you please add a comment about it too. Thanks!

Reply:
On that note, do the latest version of HostGA contain all the attributes that we're parsing? Or will we have to have a similar check like this for other versions too in the future?

Reply:
1 more question, what's the min version where the HostGA version is available? Is it .112 or greater than that?

Reply:
* I'm guessing this check is temporary until this feature is stabilized? If so, could you please add a comment about it too. Thanks!
It will stay there for a while, probably it can be removed once we get telemetry indicating that there are no old HostGAPlugins in the wild. For the moment this check is there just to avoid generating noise. No harm if this is not removed so I'll leave it as is

* On that note, do the latest version of HostGA contain all the attributes that we're parsing? Or will we have to have a similar check like this for other versions too in the future?

We will have additional checks on the version. The purpose of this PR is mainly  to set the infrastructure needed to perform those checks.

* 1 more question, what's the min version where the HostGA version is available? Is it .112 or greater than that?

Not sure of the exact version, but Flexible version defaults to 0.0.0.0 so version 0 indicates the version field is not there

Reply:
> We will have additional checks on the version. The purpose of this PR is mainly to set the infrastructure needed to perform those checks.

I am personally not a big fan of putting checks on versions because they can be never ending. But at the same time I'm not sure of a better way to do it either so approved the PR